src/routes/middlewares/authorized-user.middleware.js

```diff
@@ -15,19 +21,50 @@ import { errorResponse, verifyToken } from '../../utils/helpers.utils';
 dotenv.config();
 
 const checkAuthorizedUser = async (req, res, next) => {
-  const token = req.headers.authorization || req.headers['x-access-token'];
+  const token =
+    req.headers.authorization ||
+    req.headers['x-access-token'] ||
+    req.params.token;
 
   if (!token) {
     return errorResponse(res, 401, 'Please provide a JWT token');
   }
-  req.user = await verifyToken(token, process.env.SECRET_KEY);
-  if (!req.user) {
+
+  //Decode token to check if it contains valid user payload;
+  const decoded = await verifyToken(token);
+
+  if (!decoded) {
     return errorResponse(
       res,
       400,
       'Token is invalid, please provide a valid token',
     );
   }
+
+  //check to ensure the token id is a valid uuid
+  const { error } = Joi.validate(decoded, tokenId, {
```

Comment:
Don't you think, you should extract this into a separate function/method to handle token validation, instead of validating token inside of a `checkAuthorizedUser` middleware.

Reply:
Thanks for your observation. I just remembered now that we have a new global helper to handle this.

Reply:
I have refactored this part like you suggested